### DISCUSSION TITLE
[Enhancement] Delete Unused Images on Imgur

### DISCUSSION THREAD

**Discussion by ozntel** [gavvvr/obsidian-imgur-plugin#19] [Ideas]
Hi,

Thanks for the plugin. I just started to use and it saves a lot of place in my vault :)

Just a small request:

I didn't check the API of Imgur. However, if there is an end point, which helps listing all images for the user images after providing the token in the request header, it would be great to add an option within the plugin to cross check all Imgur Links in the vault vs the list of images uploaded on Imgur and delete the ones, which are not referenced anymore.

I can definitely give a support with the vault scan. I have a plugin, which basically checks all of the resolved links in the vault and clear the images, which are not referenced anymore but it is only applicable for the local images:

https://github.com/ozntel/oz-clear-unused-images-obsidian

We could even go further and add a scan of all links within all markdown files (since some users might refer images in the code blocks etc and Obsidian doesn't resolve them as a link in cache) and add them them for the comparison.

Let me know your thoughts

Thanks!

**Comment by gavvvr** [comment]
Hi @ozntel. I am glad to hear that this plugin makes your life better!

Am I right that you are talking about an authenticated uploads case? And you would like to remove all of the images from remote which do not have a reference in your vault?
If so, I am not sure I understand the value this feature will bring. Why do you care about unused images on remote?

I see the following problems:
- if you have more than 1 vault (which I think most of the users have) and a single Imgur account, such cleanup performed on a single vault will remove images for all the other vaults, just because they do not belong to the vault where you performed the cleanup
- I have a single Imgur account, I can use it for other purposes than making notes in a single Obsidian vault. Such cleanup will also remove any of my non-obsidian notes

Having these 2 problems in mind, I do not see much overall value in this idea. I might be wrong at understanding you. Let me know please, if so. 

I can imagine this implemented for a very specific use-case:
- you only use single vault and you only use Imgur for your Obsidian notes
- or you have an Imgur account per Obsidian vault

**Comment by ozntel** [comment]
> Am I right that you are talking about an authenticated uploads case? 

Yes

> Why do you care about unused images on remote?

It will sound weird but I am a little bit sensitive when it comes to digital garbage since somewhere around the world we fill the servers up and they need to run more machine to be able to keep them, which also means more electricity. 

> I see the following problems:

After seeing your arguments, you are actually right. It is almost impossible to create such a cleaning function especially when you don't have images only from Obsidian or single vault. I will let you know if I come up with any better idea :)
